### PR TITLE
Filter past events related to tbtc specific actions

### PR DESCRIPTION
**Pull request by dimpar**
Ref: https://github.com/keep-network/keep-ecdsa/issues/574

Depends on: https://github.com/keep-network/keep-common/pull/58

This PR replaces temp solutions for retrieving past events:
- `SignatureSubmitted`
- `DepositRedemptionRequested`

Past events retrieval can be done by calling generated functions `Past*` by `keep-common` project.

### CODE REVIEW

**Comment by pdyraga** [comment]
Two notes:
- Let's update `go.mod` `keep-common` reference to point to `d9f6bd5e51e8bcd103c1def3082afa331a3b33fe` temporarily.
- Let's regenerate contracts in `pkg/chain/gen/contract` push the updated code in this PR - those files are included in version control.

File: pkg/chain/chain.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ type BondedECDSAKeep interface {
 	// All implementations should returns those events sorted by the
 	// block number in the ascending order.
 	PastSignatureSubmittedEvents(
-		keepAddress string,
+		keepAddress common.Address,
```
Comment (pdyraga) [inline pkg/chain/chain.go:155]: We should stick with `string` if possible. It is a generic `chain` interface and we shouldn't use types from `go-ethereum`. It is true other functions are doing that but that's wrong and let's not escalate it.

File: go.sum [hunk 2]
```diff
@@ -337,10 +337,14 @@ github.com/keep-network/keep-common v1.2.1-0.20201020114759-19c123cbd4f4 h1:Civu
...
 github.com/keep-network/keep-common v1.2.1-0.20201112133003-054c43201b4b h1:ErsUI/oSwXPUHxNq7j/ttZXn3WOn4bOCf2QJoo2gvZE=
 github.com/keep-network/keep-common v1.2.1-0.20201112133003-054c43201b4b/go.mod h1:emxogTbBdey7M3jOzfxZOdfn139kN2mI2b2wA6AHKKo=
+github.com/keep-network/keep-common v1.2.1-0.20201116121154-016d77aacdce h1:B+kei2KjTdLS7JeK6YeDFU6suzPbkCn8WiAX1VDqtA0=
+github.com/keep-network/keep-common v1.2.1-0.20201116121154-016d77aacdce/go.mod h1:emxogTbBdey7M3jOzfxZOdfn139kN2mI2b2wA6AHKKo=
 github.com/keep-network/keep-core v1.3.0 h1:7Tb33EmO/ntHOEbOiYciRlBhqu5Ln6KemWCaYK0Z6LA=
 github.com/keep-network/keep-core v1.3.0/go.mod h1:1KsSSTQoN754TrFLW7kLy50pOG2CQ4BOfnJqdvEG7FA=
 github.com/keep-network/tbtc v1.1.1-0.20201026093513-cb9246987718 h1:/ZNMBY7y6hfzCYA8mgtHnspGO26OmWV3sDehyGnqRyY=
 github.com/keep-network/tbtc v1.1.1-0.20201026093513-cb9246987718/go.mod h1:igBF2MPTFkzOdZ3gcwt8h0Zb5pZaHnij/iPZoMB9IKM=
+github.com/keep-network/tbtc v1.1.1-0.20201116122054-001d3ad0cfc6 h1:p3t1G0aXXl2dZ+hVRFwSlPqXdFr3TW/dcYtKN9u79ds=
+github.com/keep-network/tbtc v1.1.1-0.20201116122054-001d3ad0cfc6/go.mod h1:OTDuVugn5XnKPpKgWNDgD8Ziv4PTUZ9mKM8k40JSl1w=
```
Comment (pdyraga) [inline go.sum:347]: Let's remember to `go mod tidy` at the end.

File: pkg/chain/ethereum/ethereum.go [hunk 3]
```diff
@@ -516,24 +514,20 @@ func (ec *EthereumChain) GetOpenedTimestamp(keepAddress common.Address) (time.Ti
 // for the given keep which occurred after the provided start block.
 // Returned events are sorted by the block number in the ascending order.
 func (ec *EthereumChain) PastSignatureSubmittedEvents(
-	keepAddress string,
+	keepAddress common.Address,
 	startBlock uint64,
 ) ([]*eth.SignatureSubmittedEvent, error) {
-	if !common.IsHexAddress(keepAddress) {
-		return nil, fmt.Errorf("invalid keep address: [%v]", keepAddress)
-	}
```
Comment (pdyraga) [inline pkg/chain/ethereum/ethereum.go:524]: We should accept `string` and this validation should stay.

File: pkg/chain/chain.go (hunk 1 above)
Comment (dimpar) [reply pkg/chain/chain.go]: Changed, as you mentioned, wanted to stay consisted with other functions and didn't realize we went against the generic interface.

File: go.sum (hunk 2 above)
Comment (dimpar) [reply go.sum:341]: Done

File: pkg/chain/ethereum/ethereum.go (hunk 3 above)
Comment (dimpar) [reply pkg/chain/ethereum/ethereum.go]: Done.

File: go.mod [hunk 4]
```diff
@@ -18,9 +18,9 @@ require (
 	github.com/gogo/protobuf v1.3.1
 	github.com/google/gofuzz v1.1.0
 	github.com/ipfs/go-log v1.0.4
-	github.com/keep-network/keep-common v1.2.1-0.20201112133003-054c43201b4b
+	github.com/keep-network/keep-common v1.2.1-0.20201116121154-016d77aacdce
```
Comment (pdyraga) [inline go.mod:21]: Let's update it to merge commit `8af0579`.

File: go.mod [hunk 5]
```diff
@@ -18,9 +18,9 @@ require (
...
 	github.com/keep-network/keep-core v1.3.0
-	github.com/keep-network/tbtc v1.1.1-0.20201026093513-cb9246987718
+	github.com/keep-network/tbtc v1.1.1-0.20201116122054-001d3ad0cfc6
```
Comment (pdyraga) [inline go.mod:23]: Let's update it to merge commit `986e007`.

File: pkg/chain/ethereum/ethereum.go [hunk 6]
```diff
@@ -519,21 +517,19 @@ func (ec *EthereumChain) PastSignatureSubmittedEvents(
 	keepAddress string,
 	startBlock uint64,
 ) ([]*eth.SignatureSubmittedEvent, error) {
-	if !common.IsHexAddress(keepAddress) {
-		return nil, fmt.Errorf("invalid keep address: [%v]", keepAddress)
+	if !common.IsHexAddress(keepAddress) {	
+		return nil, fmt.Errorf("invalid keep address: [%v]", keepAddress)
```
Comment (pdyraga) [inline pkg/chain/ethereum/ethereum.go:521]: Whitespaces 😬

File: pkg/chain/ethereum/ethereum.go [hunk 7]
```diff
@@ -519,21 +517,19 @@ func (ec *EthereumChain) PastSignatureSubmittedEvents(
...
+	keepContract, err := ec.getKeepContract(common.HexToAddress(keepAddress))
 	if err != nil {
 		return nil, err
 	}
 
-	events, err := keepContractEventLog.PastSignatureSubmittedEvents(
+	// 2nd nil denotes latest block
+	events, err := keepContract.PastSignatureSubmittedEvents(
 		startBlock,
 		nil,
```
Comment (pdyraga) [inline pkg/chain/ethereum/ethereum.go:531]: Maybe we can put the comment about `// latest block` in this line?

**Review by pdyraga — COMMENTED** [review]
Four small comments and I am good assuming all jobs turn green. I'll leave the final review and testing to @lukasz-zimnoch.

File: pkg/chain/ethereum/ethereum.go (hunk 6 above)
Comment (dimpar) [reply pkg/chain/ethereum/ethereum.go]: Removed.

File: pkg/chain/ethereum/ethereum.go (hunk 7 above)
Comment (dimpar) [reply pkg/chain/ethereum/ethereum.go]: Sure.

File: go.mod (hunk 4 above)
Comment (dimpar) [reply go.mod]: Done.

File: go.mod (hunk 5 above)
Comment (dimpar) [reply go.mod]: Done, but I will again change it to latest once [PR](https://github.com/keep-network/tbtc/pull/744) is merged to master

**Comment by dimpar** [comment]
Simple test to see if we can retrieve events and that it contains data:
- `DepositRedemptionRequested`:

```
2020-11-17T12:41:49.003+0100	INFO	tbtc-extension	starting [provide redemption proof] monitoring for deposit [0x5967B93dFeEEE73087A92A199821Dd5a79dda6c2]
PastDepositRedemptionRequestedEvents: [&{[89 103 185 61 254 238 231 48 135 169 42 25 152 33 221 90 121 221 166 194] [214 176 161 202 143 6 65 185 126 254 192 241 237 115 215 46 88 179 143 165] [176 151 244 225 43 180 134 29 34 222 113 166 174 94 126 38 80 49 188 75 236 13 112 45 144 50 241 21 21 251 34 174] 1000000 [22 0 20 173 173 49 96 135 116 89 59 18 196 92 12 143 226 99 138 184 186 167 246] 2700 [170 26 41 4 238 46 38 181 203 14 215 165 31 233 11 69 186 5 210 128 14 87 184 209 188 250 113 122 9 70 255 207 0 0 0 0] {[249 228 186 180 251 67 10 113 32 74 144 62 243 47 74 43 36 222 47 199] [[121 89 195 128 23 64 97 162 26 59 168 2 67 160 50 186 156 209 13 200 189 23 54 215 232 53 201 78 151 163 90 152] [0 0 0 0 0 0 0 0 0 0 0 0 89 103 185 61 254 238 231 48 135 169 42 25 152 33 221 90 121 221 166 194] [0 0 0 0 0 0 0 0 0 0 0 0 214 176 161 202 143 6 65 185 126 254 192 241 237 115 215 46 88 179 143 165] [176 151 244 225 43 180 134 29 34 222 113 166 174 94 126 38 80 49 188 75 236 13 112 45 144 50 241 21 21 251 34 174]] [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 15 66 64 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 128 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 10 140 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 192 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 23 22 0 20 173 173 49 96 135 116 89 59 18 196 92 12 143 226 99 138 184 186 167 246 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 36 170 26 41 4 238 46 38 181 203 14 215 165 31 233 11 69 186 5 210 128 14 87 184 209 188 250 113 122 9 70 255 207 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 2697 [79 123 163 28 191 29 152 20 17 196 132 189 29 50 159 176 113 240 207 254 35 173 188 190 159 22 26 226 229 151 31 151] 0 [141 232 11 73 178 72 229 40 73 210 249 80 126 96 99 164 184 202 30 20 223 99 86 151 174 92 231 26 246 121 65 246] 5 false}}]
```

**Comment by lukasz-zimnoch** [comment]
LGTM! I'll test these changes locally to make sure everything works. @dimpar please fix the failing build so we can merge.

**Review by lukasz-zimnoch — DISMISSED** [review]

**Review by lukasz-zimnoch — APPROVED** [review]

**Pull request merged**